src/vinylcontrol/vinylcontrolsignalwidget.cpp

```diff
@@ -76,9 +74,7 @@ void VinylControlSignalWidget::onVinylSignalQualityUpdate(const VinylSignalQuali
 
 void VinylControlSignalWidget::resetWidget()
 {
-    if (m_imageData != nullptr) {
-        memset(m_imageData, 0, sizeof(uchar) * m_iSize * m_iSize * 4);
-    }
+    m_imageData = std::vector<unsigned char>(m_iSize * m_iSize * 4);
```

Comment:
```suggestion
    m_imageData.clear()
```
just empty the vector, doesn't reallocate but leaves the capacity unchanged.

Reply:
That's wrong. You are only allowed to access the elements up to the (initialized) size of the vector. The elements between size and capacity must not be used. Clearing the vector would set its size to 0.

Reply:
good catch, then this should be valid: 
```suggestion
    m_imageData.clear()
    m_imageData.resize(m_iSize * m_iSize * 4);
```
.clear() sets the size to 0 and then .resize defaults constructs all the elements. heap-allocation and deallocation is still avoided.

Reply:
clear() is redundant, just resize() with initialization

Reply:
If I'm reading cppreference correctly, when the new size is smaller than the current size, the vector just gets truncated, if its larger, only then will the space be initialized with the elements. So when the new size is equal to the old size, the resize operation is a no-op and will leave the old items in the container.

Reply:
Oh, you are right! If re-initialization of all elements is required then resize() alone doesn't work as expected.

But then we should better wrap consecutive clear() + resize() calls into a helper function and document it properly. If it is only used once an inline comment is sufficient.